### PR TITLE
feat(TDP-2802): preparation list sort

### PR DESCRIPTION
**Pull request by jsomsanith**
**Link to the JIRA issue**
https://jira.talendforge.org/browse/TDP-2802

**Please check if the PR fulfills these requirements**
- [x] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [x] The code coverage on new code is > 75 % for backend and > 95% for frontend
- [x] The new code does not introduce new technical issues (sonar / eslint)
- [ ] Functional tests have been performed

**Please check the browsers you've tested on**
- [ ] Chrome, Firefox, Safari, Edge, IE11
- [ ] No, that's bad, this PR should not be merged !

**(Optional) What is the current behavior?**
(Additional information to the Jira)
The search part of the list component doesn't work

**(Optional) What is the new behavior?**
(Additional information to the Jira)
Now it works :)

**(Optional) Other information**:

### CODE REVIEW

File: dataprep-webapp/src/app/components/widgets-containers/preparation-list/preparation-list-controller.js [hunk 1]
```diff
@@ -40,6 +40,26 @@ export default class PreparationListCtrl {
...
 			};
 		}
+		if (changes.sortBy) {
+			const currentValue = changes.sortBy.currentValue;
+			const sortBy = this.toolbarProps.sortBy.map((sort) => {
+				const isSelected = sort.selected;
+				const shouldBeSelected = sort.id === currentValue;
+				return isSelected === shouldBeSelected ?
+					sort :
+					{ ...sort, selected: shouldBeSelected };
```
Comment (jsomsanith) [inline dataprep-webapp/src/app/components/widgets-containers/preparation-list/preparation-list-controller.js]: I wanted to enforce immutability

File: dataprep-webapp/src/assets/config/app-settings.json [hunk 2]
```diff
@@ -49,12 +50,13 @@
       },
       "toolbar": {
         "sortBy": [
-          { "id": "name", "name": "Name", "selected": true },
-          { "id": "lastModificationDate", "name": "Date" }
+          { "id": "name", "name": "Name", "field": "name" },
+          { "id": "date", "name": "Date", "field": "creationDate" }
```
Comment (jsomsanith) [inline dataprep-webapp/src/assets/config/app-settings.json]: You're right, I'll change it to `Creation Date`

File: dataprep-webapp/src/app/components/widgets-containers/preparation-list/preparation-list-controller.js (hunk 1 above)
Comment (frassinier) [inline dataprep-webapp/src/app/components/widgets-containers/preparation-list/preparation-list-controller.js]: Ok but it pretty unreadable IMO,
```
const currentSort = sort; 
if (sort.id === currentValue) {
 currentSort.selected = true;
}
return currentSort;
```

Comment (jsomsanith) [inline dataprep-webapp/src/app/components/widgets-containers/preparation-list/preparation-list-controller.js]: Ok, understood :)
I change it to if/return with indent

**Pull request merged**